toolsrc/src/vcpkg/commands.setinstalled.cpp

```diff
@@ -26,18 +27,35 @@ namespace vcpkg::Commands::SetInstalled
         // input sanitization
         const ParsedArguments options = args.parse_arguments(COMMAND_STRUCTURE);
 
-        const std::vector<FullPackageSpec> specs = Util::fmap(args.command_arguments, [&](auto&& arg) {
+        std::vector<FullPackageSpec> specs = Util::fmap(args.command_arguments, [&](auto&& arg) {
             return Input::check_and_get_full_package_spec(
                 std::string(arg), default_triplet, COMMAND_STRUCTURE.example_text);
         });
 
+        bool keep_different_triplet = false;
+
+        if (specs.empty() && GlobalState::g_manifest_mode)
+        {
+            auto res = Paragraphs::try_load_port(paths.get_filesystem(), paths.manifest_root_dir);
+            if (auto val = res.get())
+            {
+                for (auto& dep : (*val)->core_paragraph->dependencies)
+                {
+                    auto spec = Input::check_and_get_full_package_spec(
+                        std::move(dep.name), default_triplet, COMMAND_STRUCTURE.example_text);
+                    specs.push_back(spec);
```

Comment:
```suggestion
                    specs.push_back(std::move(spec));
```

Alternately `push_back(Input::check_and_get_full_package_spec(...))` so that it's already a temporary.